### ISSUE TITLE
Fully developed GBS control software

### ISSUE THREAD

**Issue by ramapcsx2** [richardloxley/RetroMatic2000#1]
Hi,
I noticed your project while doing research on the TV5725.

You may be interested in my control software here:
https://github.com/ramapcsx2/gbs-control

It's Arduino compatible, but an ESP8266 is recommended.
The code just barely fits a regular Uno, so you may have to adapt it a little.
Using this code should give you a huge image quality boost! :)

By the way, when you had issues with the I2C, that was probably because the regular MCU was still active.
It also sends commands to the TV5725 and when the 2 clash, you get corruption.
You can deactivate the original MCU by installing the jumper on "P8".

Hope this will be useful, good luck!

**Comment by richardloxley** [comment]
That's looking like a really cool project - well done!  I'll check it out in more detail when I next work on my project.

I don't think my problems weren't to do with the jumper - the built-in MCU was already deactivated with those pins.  It looked most likely that I damaged the board when I soldered on the headers for the i2c lines.

**Comment by ramapcsx2** [comment]
Hey,
I recently saw your latest update on Youtube.
In it, you mention that my software doesn't fit onto your Arduino.
I'm sorry for that. I had already switched to ESP8266 by that point, and should have just linked you to the last working Arduino Uno commit.
So for next time, here it is :)
https://github.com/ramapcsx2/gbs-control/tree/523ca42a2084f25b09a514abf0017d63e54a1dd6

The way you got YUV working was similar to what I did, by the way ;)

**Comment by richardloxley** [comment]
Cheers!  It still might not have fitted mind you, since I'm using an 
Arduino Nano on this project.

I got there in the end though :-)

Richard

**Comment by ramapcsx2** [comment]
Yeah, you compared the register dumps and figured out the important changes. Nice one :)

The Arduino will be fine, as long as it's the right chip (ATmega328).
Maybe next year? ;p

Edit:
Oh, is this a single Arduino for all the project components?
In that case, it won't fit at all, and it'd be better to deploy the scaler part on a Wemos D1 Mini (or similar ESP8266 board).
The ESP8266 is just a more capable Arduino anyway ;)
